packages/store/addon/-private/ts-interfaces/identifier.ts

```diff
@@ -3,6 +3,12 @@
 */
 import { symbol } from '../utils/symbol';
 
+type ExistingResourceObject = import('./ember-data-json-api').ExistingResourceObject;
+type ResourceIdentifierObject = import('./ember-data-json-api').ResourceIdentifierObject;
+
+export type ResourceData = ResourceIdentifierObject | ExistingResourceObject;
+export type IdentifierBucket = 'record' | never;
```

Comment:
This is equivalent to `type IdentifierBucket = 'record'`; TypeScript throws away the `| never` because `never` is the "bottom" type, which no value can ever "inhabit". In a parallel to the way that `T | any` is always just `any`: `T | never` is always just `T`.

Based on what I see below, the idea is presumably to make `GenerationMethod` more robust? I think semantically, this ends up being equivalent to doing `bucket?: IdentifierBucket` there.